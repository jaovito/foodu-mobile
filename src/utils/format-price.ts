import {LanguagePaths} from '@translations/enumns/LanguagePaths';
import {translate} from '@translations/index';

const defaultValues = {
  language: 'en-US',
  currency: 'USD',
};

function getDefaultValues() {
  if (translate(LanguagePaths.GENERAL_LANGUAGE) === 'general.language') {
    return defaultValues;
  }

  return {
    language: translate(LanguagePaths.GENERAL_LANGUAGE),
    currency: translate(LanguagePaths.GENERAL_CURRENCY),
  };
}

export function formatPrice(price: number) {
  return price.toLocaleString(getDefaultValues().language, {
    style: 'currency',
    currency: getDefaultValues().currency,
  });
}
