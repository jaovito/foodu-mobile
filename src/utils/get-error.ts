import {AxiosError} from 'axios';

export function getError(err: Error | unknown): AxiosError {
  if (err instanceof AxiosError) {
    return err;
  }

  return err as AxiosError;
}
