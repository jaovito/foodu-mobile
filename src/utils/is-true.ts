/**
 *
 * Convert booleans values, considering the cases that I understand how true and false.
 * Exemple: Empty arrays and objects, we consider false.
 *
 * @param  param - Can be any value
 *
 * @returns if `value` it isn't an empty array, returns true
 * @returns if `value` it isn't an empty object, returns true
 * @returns if `value` it isn't an empty string, returns true
 * @returns if `value` it isn't null, returns true
 * @returns if `value` it's a non-zero number, returns true
 */
export const isTrue = (param: any): boolean => {
  if (typeof param === 'object') {
    if (param === null) {
      return false;
    } else if (param instanceof Array) {
      return param.length === 0 ? false : true;
    } else {
      return Object.entries(param).length === 0 ? false : true;
    }
  }

  return !!param;
};
