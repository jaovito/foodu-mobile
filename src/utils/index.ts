import {isTrue} from './is-true';

export {isTrue};
