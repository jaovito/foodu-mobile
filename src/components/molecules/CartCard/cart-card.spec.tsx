import React from 'react';
import {fireEvent, render} from '@test/utils/test-utils';
import {CartCard} from './';
import {mock} from '@test/mocks';

describe('<CartCard />', () => {
  it('should be defined', () => {
    const card = render(
      <CartCard
        id="1"
        counter={mock.cartCard.counter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    expect(card).toBeDefined();
  });

  it('should be able to find the title card', async () => {
    const {findByText} = render(
      <CartCard
        id="1"
        counter={mock.cartCard.counter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const title = await findByText(mock.cartCard.name);

    expect(title).toBeTruthy();
  });

  it('should be able to find the description of the card', async () => {
    const {findByText} = render(
      <CartCard
        id="1"
        counter={mock.cartCard.counter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const description = await findByText(mock.cartCard.description);

    expect(description).toBeTruthy();
  });

  it('should be able to find the price of the card', async () => {
    const {findByText} = render(
      <CartCard
        id="1"
        counter={mock.cartCard.counter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const price = await findByText(
      `$${Number(mock.cartCard.price * mock.cartCard.counter || 0).toFixed(2)}`,
    );

    expect(price).toBeTruthy();
  });

  it('should be able to increment the counter', async () => {
    const {findByTestId, findByText} = render(
      <CartCard
        id="1"
        counter={mock.cartCard.counter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const incrementButton = await findByTestId(
      mock.cartCard.ids.IncrementCounterButtonID,
    );

    fireEvent.press(incrementButton);

    const counter = await findByText('2');

    expect(counter).toBeTruthy();
  });

  it('should be able to increment the counter and check it value', async () => {
    const {findByTestId, findByText} = render(
      <CartCard
        id="1"
        counter={mock.cartCard.counter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const incrementButton = await findByTestId(
      mock.cartCard.ids.IncrementCounterButtonID,
    );

    fireEvent.press(incrementButton);

    const price = await findByText(
      `$${Number(mock.cartCard.price * 2).toFixed(2)}`,
    );
    const counter = await findByText('2');

    expect(counter).toBeTruthy();
    expect(price).toBeTruthy();
  });

  it('should be able to decrement the counter', async () => {
    const {findByTestId, findByText} = render(
      <CartCard
        id="1"
        counter={2}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const decrementButton = await findByTestId(
      mock.cartCard.ids.DecrementCounterButtonID,
    );

    fireEvent.press(decrementButton);

    const counter = await findByText('1');

    expect(counter).toBeTruthy();
  });

  it('should be able to decrement the counter and check it value', async () => {
    const {findByTestId, findByText} = render(
      <CartCard
        id="1"
        counter={2}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const decrementButton = await findByTestId(
      mock.cartCard.ids.DecrementCounterButtonID,
    );

    fireEvent.press(decrementButton);

    const price = await findByText(
      `$${Number(mock.cartCard.price * mock.cartCard.counter).toFixed(2)}`,
    );
    const counter = await findByText('1');

    expect(counter).toBeTruthy();
    expect(price).toBeTruthy();
  });

  it('should be able to reset the counter if it value is less than 0 when increment', async () => {
    const {findByTestId, findByText} = render(
      <CartCard
        id="1"
        counter={0}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const incrementButton = await findByTestId(
      mock.cartCard.ids.IncrementCounterButtonID,
    );

    fireEvent.press(incrementButton);

    const price = await findByText(
      `$${Number(mock.cartCard.price * mock.cartCard.counter).toFixed(2)}`,
    );
    const counter = await findByText('1');

    expect(counter).toBeTruthy();
    expect(price).toBeTruthy();
  });

  it('not should be able to increment the counter if it value was 99', async () => {
    const maxCounter = 99;

    const {findByTestId, findByText} = render(
      <CartCard
        id="1"
        counter={maxCounter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const incrementButton = await findByTestId(
      mock.cartCard.ids.IncrementCounterButtonID,
    );

    fireEvent.press(incrementButton);

    const price = await findByText(
      `$${Number(mock.cartCard.price * maxCounter).toFixed(2)}`,
    );
    const counter = await findByText('99');

    expect(counter).toBeTruthy();
    expect(price).toBeTruthy();
  });

  it('should be able to reset the counter if it value is less than 0 when decrement', async () => {
    const {findByTestId, findByText} = render(
      <CartCard
        id="1"
        counter={0}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const decrementButton = await findByTestId(
      mock.cartCard.ids.DecrementCounterButtonID,
    );

    fireEvent.press(decrementButton);

    const price = await findByText(
      `$${Number(mock.cartCard.price * mock.cartCard.counter).toFixed(2)}`,
    );
    const counter = await findByText('1');

    expect(counter).toBeTruthy();
    expect(price).toBeTruthy();
  });

  it('should be able to call onDelete function', async () => {
    const {findByTestId} = render(
      <CartCard
        id="1"
        counter={2}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    );

    const deleteButton = await findByTestId(
      mock.cartCard.ids.DeleteCartButtonID,
    );

    fireEvent.press(deleteButton);

    expect(mock.cartCard.onDelete).toHaveBeenCalled();
  });

  it('should be able to match with snapshot', () => {
    const card = render(
      <CartCard
        id="1"
        counter={mock.cartCard.counter}
        description={mock.cartCard.description}
        image={mock.cartCard.image}
        name={mock.cartCard.name}
        onDelete={mock.cartCard.onDelete}
        price={mock.cartCard.price}
      />,
    ).toJSON();

    expect(card).toMatchSnapshot();
  });
});
