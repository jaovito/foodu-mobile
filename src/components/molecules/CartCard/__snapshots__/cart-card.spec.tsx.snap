// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCard /> should be able to match with snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderColor": "rgba(0, 0, 0, 0.04)",
        "borderRadius": 16,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 31,
        "paddingBottom": 24,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
        "shadowColor": "rgba(0, 0, 0, 0.12)",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "height": 196,
          "marginRight": 16,
          "width": 196,
        },
      ]
    }
  >
    <Image
      onError={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "http://github.com/JohnDoe.png",
        }
      }
      style={
        [
          {
            "borderRadius": 16,
            "height": 249,
            "width": 249,
          },
          {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 100,
            "bottom": 16,
            "height": 47,
            "justifyContent": "center",
            "position": "absolute",
            "right": 16,
            "width": 47,
          },
        ]
      }
    >
      <Text
        fontWeight="bold"
        style={
          [
            {
              "color": "#0D0D0D",
              "fontFamily": "Poppins_700Bold",
              "fontSize": 31,
            },
            {
              "color": "#FFFFFF",
              "fontSize": 24,
              "lineHeight": 29,
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "90%",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      fontWeight="bold"
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_700Bold",
            "fontSize": 31,
          },
          {
            "fontSize": 35,
          },
        ]
      }
    >
      John Doe
    </Text>
    <Text
      fontWeight="regular"
      numberOfLines={3}
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {
            "fontSize": 24,
            "maxWidth": "100%",
          },
        ]
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Saepe rerum iure magni porro ipsum blanditiis consequatur, corrupti doloremque natus. Deserunt alias delectus nisi commodi maiores. Eligendi harum laborum excepturi possimus?
    </Text>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 31,
            "marginTop": 16,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0D0D0D",
              "fontFamily": "Poppins_400Regular",
              "fontSize": 31,
            },
            {},
          ]
        }
      >
        $
        20.00
      </Text>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 16,
              "justifyContent": "center",
            },
          ]
        }
      >
        <RNGestureHandlerButton
          activeOpacity={0.8}
          borderless={true}
          collapsable={false}
          delayLongPress={600}
          handlerTag={36}
          handlerType="NativeViewGestureHandler"
          onActiveStateChange={[Function]}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          onPress={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FE8C00",
              "borderRadius": 31,
              "height": 63,
              "justifyContent": "center",
              "opacity": 1,
              "width": 63,
            }
          }
          testID="DecrementCounterButtonID"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 31,
                  },
                ],
                {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          activeOpacity={0.8}
          borderless={true}
          collapsable={false}
          delayLongPress={600}
          handlerTag={37}
          handlerType="NativeViewGestureHandler"
          onActiveStateChange={[Function]}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          onPress={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FE8C00",
              "borderRadius": 31,
              "height": 63,
              "justifyContent": "center",
              "opacity": 1,
              "width": 63,
            }
          }
          testID="IncrementCounterButtonID"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 31,
                  },
                ],
                {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          activeOpacity={0.8}
          borderless={true}
          collapsable={false}
          color="red"
          delayLongPress={600}
          handlerTag={38}
          handlerType="NativeViewGestureHandler"
          onActiveStateChange={[Function]}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          onPress={
            [MockFunction] {
              "calls": [
                [],
              ],
              "results": [
                {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          style={
            {
              "alignItems": "center",
              "backgroundColor": "red",
              "borderRadius": 31,
              "height": 63,
              "justifyContent": "center",
              "opacity": 1,
              "width": 63,
            }
          }
          testID="DeleteCartButtonID"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 31,
                  },
                ],
                {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </RNGestureHandlerButton>
      </View>
    </View>
  </View>
</View>
`;
